lib/base64-util.js

```diff
@@ -0,0 +1,30 @@
+const logger = require('./logger');
+
+const atob = s => String.fromCharCode(...Buffer.from(s, 'base64'));
+const btoa = s => Buffer.from(s.split('').map(c => c.charCodeAt(0))).toString('base64');
+
+module.exports.b64EncodeUnicode = function(str) {
+    // (1) use encodeURIComponent to get percent-encoded UTF-8
+    // (2) convert percent encodings to raw bytes
+    // (3) convert raw bytes to Base64
+    try {
+        return btoa(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, (match, p1) => {
+            return String.fromCharCode('0x' + p1);
+        }));
+    } catch(e) {
+        logger.error(`b64EncodeUnicode: failed to encode ${str}`);
+        return null;
```

Comment:
Are callers prepared to handle `null`? Might be safer to return an empty string?

Reply:
they are, but fair enough - empty string it is